pkg/datapath/linux/routing/routing.go

```diff
@@ -49,17 +51,29 @@ func (info *RoutingInfo) Configure(ip net.IP, mtu int, masq bool) error {
 		}).Warning("Unable to configure rules and routes because IP is not an IPv4 address")
 		return errors.New("IP not compatible")
 	}
+
 	ifindex, err := retrieveIfIndexFromMAC(info.MasterIfMAC, mtu)
 	if err != nil {
 		return fmt.Errorf("unable to find ifindex for interface MAC: %s", err)
 	}
 
+	// Grab the table ID from the interface index. This is necessary because we
+	// need to store the endpoint's rules and routes in a specific table,
+	// per-ENI / Azure interface. If we don't put them in specifc tables, then
+	// the table ID and ifindex may collide, messing up the routing table when
+	// new rules and routes are inserted.
+	tableID, err := retrieveTableIDFromIfindex(ifindex)
+	if err != nil {
+		return fmt.Errorf("unable to find table ID from iface name: %w", err)
+	}
+
 	ipWithMask := net.IPNet{
 		IP:   ip,
 		Mask: net.CIDRMask(32, 32),
 	}
 
-	// Route all traffic to the interface address via the main routing table
+	// Route all traffic to the interface address via the main routing table.
+	// Ingress rules are an exception to the above comment.
```

Comment:
I can't find the historical context on why ingress rules need to be in the main routing table. 

Side note, if we decide to change it to be the same as the egress rule, then we'd have to change the protoype of `Delete()` which only accepts an IP. We wouldn't have enough information to find out which table the ingress rule is in because we'd need the iface name to retrieve the table ID. Just wanted to throw this out there in case we go down this route.